src/includes/functions.php

```diff
@@ -322,3 +206,114 @@ function get_increase_time_limit_max()
     global $_increase_time_limit_max;
     return $_increase_time_limit_max;
 }
+
+
+/**
+ * Returns the temporary folder path that silverstripe should use for its cache files.
+ *
+ * @param string $base The base path to use for determining the temporary path
+ * @return string Path to temp
+ */
+function getTempFolder($base = null)
```

Comment:
There 3 functions can be probably be pushed into a single class with 3 static methods, of which only the first one (this one) is public. `TempFolderCreator` or something (since it creates as well as finds the temp folder).

Benefits:
 * cleaner root namespace
 * smaller public API (only the first method)
 * can use autoloading